lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -868,9 +881,10 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
                 )
             )
             self._post_start_init(event)
-        except (OpenSearchStartTimeoutError, OpenSearchNotFullyReadyError):
+        except (OpenSearchHttpError, OpenSearchStartTimeoutError, OpenSearchNotFullyReadyError):
             event.defer()
-        except OpenSearchStartError as e:
+        except (OpenSearchStartError, OpenSearchUserMgmtError) as e:
+            # Either generic start failure or cluster is not read to create the internal users
```

Comment:
when would the cluster not be ready to create the users?

Reply:
If we do not have enough peers to form a cluster? E.g. this is the first unit booting up